### PR TITLE
ENH: Re-implement OpenIGTLinkRemote support for IMGMETA/LBMETA messages

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
relies on https://github.com/IGSIO/OpenIGTLinkIO/pull/76

### CODE REVIEW

File: OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx [hunk 1]
```diff
@@ -353,12 +366,76 @@ void vtkMRMLIGTLConnectorNode::vtkInternal::ProcessIncomingDeviceModifiedEvent(
...
     }
   }
 
+  vtkMRMLIGTLQueryNode* queryNode = this->GetQueryNodeForDevice(modifiedDevice);
+  if (queryNode && modifiedNode)
+  {
+    queryResponses.push_back(queryNode);
+    queryNode->SetResponseDataNodeID(modifiedNode->GetID());
+  }
+  this->RemovePendingQuerynode(queryNode);
```
Comment (lassoan) [inline OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx]:
```suggestion
  this->RemovePendingQueryNode(queryNode);
```

File: OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx [hunk 2]
```diff
@@ -92,6 +100,9 @@ class vtkMRMLIGTLConnectorNode::vtkInternal
 
   void DeviceAboutToReceiveEvent(igtlioDevice* modifiedDevice);
 
+  vtkMRMLIGTLQueryNode* GetQueryNodeForDevice(igtlioDevice* device);
+  bool RemovePendingQuerynode(vtkMRMLIGTLQueryNode* queryNode);
```
Comment (lassoan) [inline OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx]: Add short documentation of what this method does.

File: OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx [hunk 3]
```diff
@@ -353,12 +366,76 @@ void vtkMRMLIGTLConnectorNode::vtkInternal::ProcessIncomingDeviceModifiedEvent(
...
+        }
+      }
     }
   }
 
+  vtkMRMLIGTLQueryNode* queryNode = this->GetQueryNodeForDevice(modifiedDevice);
+  if (queryNode && modifiedNode)
+  {
+    queryResponses.push_back(queryNode);
+    queryNode->SetResponseDataNodeID(modifiedNode->GetID());
```
Comment (lassoan) [inline OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx]: We set this a couple of lines below. Is it necessary to set it here, too?

File: OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx [hunk 4]
```diff
@@ -353,12 +366,76 @@ void vtkMRMLIGTLConnectorNode::vtkInternal::ProcessIncomingDeviceModifiedEvent(
...
+  }
+  this->RemovePendingQuerynode(queryNode);
+  for (QueryListType::iterator queryResponseIt = queryResponses.begin(); queryResponseIt != queryResponses.end(); ++queryResponseIt)
+  {
+    vtkMRMLIGTLQueryNode* queryNode = *queryResponseIt;
+    queryNode->SetQueryStatus(vtkMRMLIGTLQueryNode::STATUS_SUCCESS);
```
Comment (lassoan) [inline OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx]: Can a query time out? I think timeout feature existed before but I don't see it now.

Comment (Sunderlandkyl) [reply OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx]: Ah, I meant to add the time out as well.

**Pull request merged**